lua/nui/tree/init.lua

```diff
@@ -344,6 +344,11 @@ end
 ---@return NuiTreeNode
 function Tree:remove_node(node_id)
   local node = self.nodes.by_id[node_id]
+  for _, child_id in ipairs(node._child_ids) do
+    -- We might want to store the nodes and return them with the node itself?
+    -- We should _really_ not be doing this recursively, but it will work for now
+    Tree:remove_node(child_id)
```

Comment:
If the task of removing node from `self.nodes.by_id` is moved to a separate local function, we can avoid doing `vim.tbl_filter` on `node._child_ids` for each `child_id`. Something like:

```lua
local function remove_node(tree, node_id)
  -- remove node with the children recursively from tree.nodes.by_id` 
end
```

And then use it as:

```lua
function Tree:remove_node(node_id)
  local node = remove_node(self, node_id)
  -- rest of the function
end
```

Reply:
That makes sense. I migrated the node removal logic to its own local function and updated the PR to match